### PR TITLE
Reader for scienta .ibw (Igor binary wave) data

### PR DESCRIPTION
**Pull request by lukaspie**
- [x] adds a parser for Scienta .ibw data
- [x] harmonize such that the same config and mapper can be used for all scienta parsers
- [X] update examples  

**This PR depends on #47 (contains the changes from there already) and should thus be reviewed/merged afterwards.** EDIT: #47 was merged.

### CODE REVIEW

File: pynxtools_xps/value_mappers.py [hunk 1]
```diff
@@ -44,7 +45,11 @@
     "No": False,
 }
 
-INTENSITY_UNIT_MAP = {"counts/s": "counts per second", "CPS": "counts per second"}
+INTENSITY_UNIT_MAP = {
+    "Counts": "counts",
+    "counts/s": "counts per second",
+    "CPS": "counts per second",
```
Comment (RubelMozumder) [inline pynxtools_xps/value_mappers.py:51]: Will the units `counts per second` be parsed in Nomad? I am not sure.

**Review by RubelMozumder — COMMENTED** [review]
I already started to review. By tomorrow it will be finished. I need to check some functions.

File: pynxtools_xps/value_mappers.py (hunk 1 above)
Comment (lukaspie) [reply pynxtools_xps/value_mappers.py:51]: You are right, I changed it to `counts_per_second`, which should be allowed: https://gitlab.mpcdf.mpg.de/nomad-lab/nomad-FAIR/-/blob/develop/nomad/units/default_en.txt?ref_type=heads#L181

File: pynxtools_xps/scienta/scienta_mappings.py [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+)
+
+
+def _extract_energy_units(energy_units: str):
+    """
+    Extract energy units from the strings for energy_units.
+    Binding Energy [eV] -> eV
+
+    """
+    return re.search(r"\[(.*?)\]", energy_units).group(1)
```
[the remark this answers is not in the document]
Comment (domna) [reply pynxtools_xps/scienta/scienta_mappings.py:30]: I would check for None here to not fail completely if the unit is not found. Maybe there is also a default which can be given to `_extract_energy_units`?

File: pynxtools_xps/scienta/scienta_mappings.py [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+
+
+def _separate_dimension_scale(scale: str):
```
Comment (domna) [inline pynxtools_xps/scienta/scienta_mappings.py:33]:
```suggestion
def _separate_dimension_scale(scale: str) -> np.typing.ArrayLike:
```

File: pynxtools_xps/scienta/scienta_reader.py [hunk 4]
```diff
@@ -528,67 +403,207 @@ def _parse_region(self, region_id):
...
+
+        notes = self._parse_note(wave["note"])
+
+        data_unit_label, data_unit = self._parse_unit(wave["data_units"])
+        dimension_unit_label, dimension_unit = self._parse_unit(wave["dimension_units"])
+
+        wave_header = wave["wave_header"]
+        data = wave["wData"]
+
+        # Not needed at the moment
```
Comment (domna) [inline pynxtools_xps/scienta/scienta_reader.py:447]: Maybe it's good to state here why it is kept. Something like `TODO: Add support for formulas` or so

File: pynxtools_xps/scienta/scienta_reader.py [hunk 5]
```diff
@@ -528,67 +403,207 @@ def _parse_region(self, region_id):
...
-        try:
-            [key, value] = line.split("=")
-            key = convert_pascal_to_snake(key)
+        unit = bunit.decode("utf-8").replace("\r", "\n")
 
-            key = KEY_MAP.get(key, key)
-            value = _re_map_single_value(key, value, VALUE_MAP)
+        pattern = r"([\w\s]+)\s*\[([\w\s.]+)\]"
+        matches = re.match(pattern, unit)
+        if matches:
```
Comment (domna) [inline pynxtools_xps/scienta/scienta_reader.py:558]:
```suggestion
        if matches is not None:
```

**Review by domna — APPROVED** [review]
Some small suggestions. Otherwise good to go

File: pynxtools_xps/scienta/scienta_mappings.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (lukaspie) [reply pynxtools_xps/scienta/scienta_mappings.py:30]: changed so the default is eV

File: pynxtools_xps/scienta/scienta_mappings.py (hunk 3 above)
Comment (lukaspie) [reply pynxtools_xps/scienta/scienta_mappings.py:33]: added

File: pynxtools_xps/scienta/scienta_reader.py (hunk 5 above)
Comment (lukaspie) [reply pynxtools_xps/scienta/scienta_reader.py:558]: changed

File: pynxtools_xps/reader_utils.py [hunk 6]
```diff
@@ -352,6 +352,31 @@ def _re_map_single_value(
...
+        For testing, this can be a str or a np.ndarray.
+
+    Returns
+    -------
+    bool
+        True if the string or np.ndarray is not empty.
+
+    """
+    for datatype in [str, int, float]:
+        if isinstance(value, datatype) and value:
```
[the remark this answers is not in the document]
Comment (lukaspie) [reply pynxtools_xps/reader_utils.py:371]: Yeah, that makes more sense.

**Pull request merged**